Fix up double-conversion and harfbuzz

tronical opened the pull request:
This pull request fixes the double-conversion build on Linux (implicitly, through bumping to upstream that has fixed cmake targets) and also fixes external harfbuzz usage.

tronical committed:
[double-conversion] Bump to 3.1.0

This release also allows removing the workarounds for calling
vcpkg_fixup_cmake_targets twice after renaming *LibraryDepends.cmake
because upstream didn't install *Targets.cmake files. That in turn fixes
using double-conversion on Linux.

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Microsoft/vcpkg?pullRequest=4285) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Microsoft/vcpkg?pullRequest=4285) before we can accept your contribution.<br/><br/>:x: tronical [sign now](https://cla.opensource.microsoft.com/Microsoft/vcpkg?pullRequest=4285)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Microsoft/vcpkg?pullRequest=4285) it.</sub>

tronical committed:
[harfbuzz] Fix cmake configs

When building cmake with autotools, a harfbuzz-config.cmake file gets
installed that makes find_package work. However when building with
cmake, that is omitted. Therefore include a patch from upstream pull
request

    https://github.com/harfbuzz/harfbuzz/pull/1161

to fix that.

ras0219-msft committed:
[harfbuzz] Rename targets to unofficial-harfbuzz because they are not provided by upstream

ras0219-msft commented:
Thanks for the PR!

Since these targets aren't provided by upstream harfbuzz, I named them `unofficial::harfbuzz::` so they will not conflict with any targets that harfbuzz wants to eventually provide.
